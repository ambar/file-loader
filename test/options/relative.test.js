/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('relative', () => {
    test('{Boolean} - `false`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            relative: false,
          },
        },
      };

      const stats = await webpack('nested/fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Boolean} - `true`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            relative: true,
          },
        },
      };

      const stats = await webpack('nested/fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Boolean} - `true` with relative `context`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            context: './test/fixtures/nested/',
            relative: true,
          },
        },
      };

      const stats = await webpack('nested/fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Boolean} - `true` with absolute `context`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            context: '../nested/',
            relative: true,
          },
        },
      };

      const stats = await webpack('nested/fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });
  });
});
