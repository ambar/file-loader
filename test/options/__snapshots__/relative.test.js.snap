// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options relative {Boolean} - \`false\` 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options relative {Boolean} - \`true\` 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"../9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options relative {Boolean} - \`true\` with absolute \`context\` 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options relative {Boolean} - \`true\` with relative \`context\` 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
